charts/opentelemetry-kube-stack/templates/collector.yaml

```diff
@@ -113,6 +113,7 @@ spec:
   {{- toYaml . | nindent 4}}
   {{- end }}
   volumeMounts:
+  {{- if or ($collector.presets.logsCollection.enabled) ($collector.presets.logsCollection.storeCheckpoints) ($collector.presets.hostMetrics.enabled) ($collector.volumeMounts) }}
```

Comment:
I think this needs to be placed above `volumeMounts` so that we only add the section if there is something that will be mounted.